### PR TITLE
fix: Remotes not persisted

### LINKED ISSUE #4441: Remotes not persisted
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
Since 2.50 whenever remote URL or Push URL added or modified, they are no longer persisted in the config.
The regression was introduced during the refactor of the `FormRemotes` in 613335552bf0735080e442629da1b71b99551d53

**What is the expected behavior?**
Restore the ability to save new or changed remotes by ensuring they are added to the collection whenever 'Save' button is clicked.

A use case for deletion of remotes requires further thoughts.
Deleting a remote from the history list may be undesirable as it would hinder user's ability to *quickly* clone the remote repository.
The flipside is that the history list may grow long without a UI to manage it.
This issue should be considered/addressed separately.

**Did this work in previous version of GitExtensions (which)?**
Presumably 2.49

### PR DESCRIPTION
**Pull request by RussKie**
Since 2.50 user remotes are no longer persisted in the config.
The regression was introduced during the refactor of the `FormRemotes` in 613335552bf0735080e442629da1b71b99551d53

Restore the ability to save new or changed remotes by ensuring they are added to the collection whenever 'Save' button is clicked.
This will allow saving remotes for new and currently configured remotes.

Deletion of remotes is still absent and requires further thoughts.
Deleting a remote from the history list may be undesirable as it would hinder user's ability to *quickly* clone the remote repository.
The flipside is that the history list may grow long without a UI to manage it.

Fixes #4441

What did I do to test the code and ensure quality:
 - manual testing

### CODE REVIEW

File: GitUI/CommandsDialogs/FormRemotes.cs [hunk 1]
```diff
@@ -281,51 +310,64 @@ private void btnToggleState_Click(object sender, EventArgs e)
...
+                                                       PuttySshKey.Text);
 
                 if (!string.IsNullOrEmpty(result.UserMessage))
                 {
                     MessageBox.Show(this, result.UserMessage, _gitMessage.Text);
                 }
+                else
+                {
+                    var remotes = Repositories.RemoteRepositoryHistory.Repositories;
+                    RemoteUpdate(remotes, _selectedRemote.Url, remoteUrl);
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormRemotes.cs]: I got an exception when adding a repo here, _selectedRemote was null. Not investigated

**Comment by RussKie** [comment]
Hmm, thank you. Obviously I missed some use case...
Can you describe your scenario?

File: GitUI/CommandsDialogs/FormRemotes.cs (hunk 1 above)
Comment (RussKie) [reply GitUI/CommandsDialogs/FormRemotes.cs]: I obviously didn't try to add a new remote.
Should be fixed now - tried all of them: add new, rename, delete

Comment (gerhardol) [reply GitUI/CommandsDialogs/FormRemotes.cs]: Yes, seem to be working now.

**Comment by gerhardol** [comment]
Unrelated to this correction, just writing down while I remember (could be an issue but there are so many already).

ConfigureRemotes() is slow with a few remotes, refs are fetched multiple times:

        public void ConfigureRemotes(string remoteName)
        {
            var localConfig = _module.LocalConfigFile;

            foreach (var remoteHead in _module.GetRefs(true, true))
            {
                foreach (var localHead in _module.GetRefs(true, true))
                {

**Pull request merged**